backend/src/main/java/gov/cdc/usds/simplereport/api/organization/OrganizationMutationResolver.java

```diff
@@ -79,23 +70,77 @@ public ApiFacility addFacility(
       List<String> deviceIds,
       List<UUID> deviceSpecimenTypes,
       String defaultDeviceId) {
-    _os.assertFacilityNameAvailable(testingFacilityName);
 
-    // For historical reasons, entities representing the internal database ID of a device type
-    // may be typed in the GraphQL schema as a String instead of an ID.
-    // Standardize on the UUID type for these IDs downstream of the mutation
-    List<UUID> deviceInternalIds =
-        deviceIds.stream().map(UUID::fromString).collect(Collectors.toList());
-    UUID defaultDeviceInternalId = UUID.fromString(defaultDeviceId);
+    List<UUID> deviceIdsFromDeviceSpecimenTypes =
```

Comment:
This mapping is neat! It does look like it's copied below in `updateFacility` though, could you break it out to a helper method? The logic is a bit complex so I want to make sure we're not changing it in one location and not the other down the line.

Reply:
Great catch!